graders/c/cgrader/cgrader.py

```diff
@@ -113,10 +142,23 @@ def compile_file(
         objs = []
         for std_c_file in c_file if isinstance(c_file, list) else [c_file]:
             obj_file = re.sub(r"\.[^.]*$", "", std_c_file) + ".o"
+            std_obj_files.append(obj_file)
             out += self.run_command(
                 [compiler, "-c", std_c_file, "-o", obj_file] + cflags, sandboxed=False
             )
-            std_obj_files.append(obj_file)
+            # Identify references to functions intended to disable sanitizers from object file
+            if os.path.isfile(obj_file):
+                symbols = self.run_command(
+                    ["nm", "-j", obj_file], sandboxed=False
+                ).splitlines()
+                found_symbols = [s for s in INVALID_SYMBOLS if s in symbols]
```

Comment:
Done in c61325073. This issue of empty lists/sets is always confusing, since JS and Python assign different boolean interpretations for them (in JS an empty list/dict is true).